hardhat.config.ts

```diff
@@ -0,0 +1,20 @@
+import { HardhatUserConfig } from "hardhat/types";
+
+import '@typechain/hardhat'
+import "@nomiclabs/hardhat-ethers";
+import "@nomiclabs/hardhat-waffle";
+
+const config: HardhatUserConfig = {
+  defaultNetwork: "hardhat",
+  solidity: {
+    compilers: [{ version: "0.7.3", settings: {} }],
```

Comment:
add `optimizer: {enabled: true, runs: 200},` to settings

Reply:
We can use a more recent compiler version, but we can wait until we know what Solidity dependencies we have before deciding on a version